packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
+
+type DataOnly<T> = Omit<T, FilteredKeys<T, Function>>;
+type FunctionOnly<T> = Pick<T, FilteredKeys<T, Function>> &
+  (T extends Function ? (...args: Parameters<T>) => ReturnType<T> : {});
+
+interface Remotable<T> {
+  __Remote__: T;
+}
+type Remote<Primary, Local = DataOnly<Primary>> = ERef<
+  Local & Remotable<Primary>
+>;
```

Comment:
Sorry, I'm just not getting these. From the terminology we've been using, I have an expectation of what `Remote<T>` or `Remotable<T>` would mean. But I haven't been able to fit any hypothesis into the declarations written here.

Reply:
`Remotable<T>` is confusing and should probably be written `RemoteIntefaceOf<T>`.

`Remote<T>` creates a type that accepts both near and marshalled references that were returned from `Remotable` or `Far`.